### ISSUE TITLE
Accessibility of this book

### ISSUE THREAD

**Issue by evilditch** [mml-book/mml-book.github.io#616]
Hi,

This book is our course book for the University of Helsinki's course of mathematics of  machine learning. I'm blind, and at first I was glad, because your book is available in digital form. Unfortunately, the PDF file for the book is not accessible: it has an incorrect reading order, and the images have no text alternatives.

Is it possible for me to find the bookin plain text format, such as latex code?

**Comment by mpd37** [comment]
Thanks for reaching out. Unfortunately, we aren't allowed to share the LaTeX sources. However, we'd like to work on the accessibility of the book. We tried in the past to use the accessibility package, but this doesn't work with the (brittle) template we use. I believe that some problems may arise from the fact that the template is some form of (customized) tufte style (we use the margins quite a bit). 

Maybe we can work together on fixing the reading order and some other issues that you run into?

**Comment by vishwesh5** [comment]
Hi, just came across this issue. Please let me know if I can also contribute to this somehow.

**Comment by evilditch** [comment]
@mpd37 Thanks for your answer!

I have tried to use propably same accessibility package, but it does not work at all. Currently I'm studying more about latex packages and accessible PDF files, so I would fix the accessibility package problems.

In reality, PDF is not a very good file format for screen reader users, even if it should be as accessible as possible.

I think, can you convert the book to HTML and mathml? then screen reader can read math symbols better than now, and the headings are semantically marked.

(For the course, I would need chapters 1 to 4. 😄)

**Comment by chengsoonong** [comment]
I tried a few things:
- using lualatex to compile to PDF
- apply "Accessibility wizard" in Acrobat Pro to generate a searchable image.

Using the Acrobat screen reader, the maths is still pretty mangled in Chapter 4. The issue seems to be subscripts and matrix equations.

**Comment by chengsoonong** [comment]
I also tried using pandoc naively on the latex sources to generate html, but got quickly bogged down with error messages. Happy to try more if someone can point me to a guide on pandoc on latex.

**Comment by evilditch** [comment]
I tried to generate mathml from our lecture slides (.tex file with matrices and other math equations) by a few various programs. The best result comes from tex4ht:
$ mk4ht mzlatex input.tex

**Comment by chengsoonong** [comment]
I had another go at using htlatex today, and did not manage to compile a sensible looking HTML document. We seem to be using some packages that make it difficult for htlatex. Even after commenting them out (they seemed to be PDF placement and referencing packages) there was an error which I have not yet managed to resolve. So not a particularly useful update.

**Comment by adimitr** [comment]
Could I ask for an update to this request. We are planning to use the book in an upcoming class in the fall and our system just flagged it as non-accessible as well. Any notes and thoughts would be appreciated.
Related: do you know if that CUP has an accessible version? We could buy those for a few students, while using the version here for everybody else.
Thanks
Alex
